### PR TITLE
student notification viewing

### PR DESCRIPTION
**Pull request by emmaeagles**
# Summary

student notification bell corresponds to their unseens notifications and when you click on it you go to a page with all your notifications on it 

## Test Plan

How did you test your changes?

i looked at it 

## Related Issues

#104

### CODE REVIEW

File: Frontend/src/components/student/NotificationListItem.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <div>
+    <div class="row items-center">
+      <div class="col">
+        <div>
+          <b>From {{ notif.admin.firstName }} {{ notif.admin.lastName }}</b>
+        </div>
+        <div>need to get date</div>
+        <div>{{ notif.title }}</div>
+        <div>{{ notif.body }}</div>
+      </div>
```
Comment (emmaeagles) [inline Frontend/src/components/student/NotificationListItem.vue]: Lol i need to figure out how we want to do this

**Comment by kraglalbert** [comment]
Can you add some screenshots for how things look

File: Frontend/src/components/student/NotificationsList.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+<script>
+import NotificationListItem from "components/student/NotificationListItem.vue";
+
+export default {
+  components: {
+    NotificationListItem
+  },
+  data() {
+    return {
+      notfications: [],
```
Comment (kraglalbert) [inline Frontend/src/components/student/NotificationsList.vue]: typo in variable name

File: Frontend/src/components/student/NotificationsList.vue [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<template>
```
Comment (kraglalbert) [inline Frontend/src/components/student/NotificationsList.vue]: Is this on the separate notifications page or the list the shows up when you click the bell?

File: Backend/src/main/java/ca/mcgill/cooperator/controller/NotificationController.java [hunk 4]
```diff
@@ -83,6 +84,32 @@ public NotificationDto getNotificationByTitle(@RequestParam String title) {
...
+
+        return ControllerUtils.convertNotifListToDto(n);
+    }
+    
+    /**
+     * Get all Notifications for student
+     *
+     * @return List<NotificationDto>
+     */
+    @GetMapping("/all/{id}")
```
[the remark this answers is not in the document]
Comment (emmaeagles) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/NotificationController.java]: the default get with an id returns a specific notification but i want all notifications of a student here

File: Frontend/src/components/student/NotificationsList.vue (hunk 3 above)
Comment (emmaeagles) [reply Frontend/src/components/student/NotificationsList.vue]: @kraglalbert this is a seperate page that appears when you click the bell

File: Backend/src/main/java/ca/mcgill/cooperator/controller/NotificationController.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (kraglalbert) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/NotificationController.java]: Oh my bad, I would make it `/student/{id}` then

**Comment by emmaeagles** [comment]
image

**Comment by emmaeagles** [comment]
image
 
^ Notification page with a new notification

**Comment by kraglalbert** [comment]
LGTM 🏖

**Pull request merged**